core/src/ledger/storage/mod.rs

```diff
@@ -1282,6 +1285,54 @@ pub mod testing {
             }
         }
     }
+
+    impl TestStorage {
+        /// Initializes all tokens in the test storage
+        pub fn initalize_tokens(
+            &mut self,
+            total_supply: token::Amount,
+            total_supply_in_masp: token::Amount,
+        ) {
+            let masp_rewards = address::masp_rewards();
+            let masp_addr = address::masp();
+            for addr in masp_rewards.keys() {
+                parameters::Parameters::init_storage(
+                    &parameters::Parameters::default(),
+                    addr,
+                    self,
+                );
+                let initial_inflation: u64 = 1;
+                let initial_locked_ratio: Decimal = dec!(0.1);
+
+                StorageWrite::write(
+                    self,
+                    &token::last_inflation(addr),
+                    initial_inflation,
+                )
+                .expect("Should not fail to put a test inflation source");
+                StorageWrite::write(
+                    self,
+                    &token::last_locked_ratio(addr),
+                    initial_locked_ratio,
+                )
+                .expect("Should not fail to put a test inflation source");
+
+                self.write(
+                    &token::total_supply_key(addr),
+                    total_supply.try_to_vec().unwrap(),
+                )
+                .expect("Should not fail to put a test total supply");
+                self.write(
+                    &token::balance_key(addr, &masp_addr),
+                    total_supply_in_masp.try_to_vec().unwrap(),
+                )
```

Comment:
I'm not sure how this gets initialized on chain from the genesis config @brentstone @murisi would either of you know?